### PR TITLE
Message send flow docs

### PR DESCRIPTION
**Pull request by A-Shumway42**
## Description

This PR adds docs in `all.md` explaining the general path of the message send flow. Code comments have been added in the logic to help explain the process.

## Security Considerations

None

### CODE REVIEW

File: app/celery/provider_tasks.py [hunk 1]
```diff
@@ -91,6 +91,13 @@ def check_sms_delivery_receipt(self, message_id, notification_id, sent_at):
     bind=True, name="deliver_sms", max_retries=48, default_retry_delay=300
 )
 def deliver_sms(self, notification_id):
+    """
+    This logic will branch off to the final step in delivering
+    the notification to sns.
```
Comment (xlorepdarkhelm) [inline app/celery/provider_tasks.py:96]: Not to be pedantic, but can these be conformed to PEP-257?

The first line is a complete sentence (not broken up over multiple lines). There needs to be a separate blank line after the first line. It also needs to be an active, direct statement, not merely a description of what it does.

`Branch off to the final step in delivering the notification to sns.` would be a more compliant docstring first line.

File: app/celery/tasks.py [hunk 2]
```diff
@@ -31,6 +31,10 @@
 
 @notify_celery.task(name="process-job")
 def process_job(job_id, sender_id=None):
+    """
+    We update job status, get the csv data from s3 and begin processing
+    csv rows.
+    """
```
Comment (xlorepdarkhelm) [inline app/celery/tasks.py:37]: PEP-257 would have this be single-line, which does mean the triple quotes are also on the same line.

`"""Update job status, get csv data from s3, and begin processing csv rows."""`

File: app/celery/tasks.py [hunk 3]
```diff
@@ -109,6 +114,10 @@ def get_recipient_csv_and_template_and_sender_id(job):
 
 
 def process_row(row, template, job, service, sender_id=None):
+    """
+    The process will branch off based on notification type,
+    sms or email.
+    """
```
Comment (xlorepdarkhelm) [inline app/celery/tasks.py:120]: PEP-257

File: app/celery/tasks.py [hunk 4]
```diff
@@ -163,7 +175,14 @@ def __total_sending_limits_for_job_exceeded(service, job, job_id):
 
 @notify_celery.task(bind=True, name="save-sms", max_retries=5, default_retry_delay=300)
 def save_sms(self, service_id, notification_id, encrypted_notification, sender_id=None):
+    """
+    We are persisting the notification to the db here and
+    placing notification in queue to send to sns.
+    """
```
Comment (xlorepdarkhelm) [inline app/celery/tasks.py:181]: PEP-257

File: app/delivery/send_to_providers.py [hunk 5]
```diff
@@ -22,6 +22,11 @@
 
 
 def send_sms_to_provider(notification):
+    """
+    This is the last step in the message send flow.
+    We get necessary data for recipient, template,
+    notification and send it off to the provider.
+    """
```
Comment (xlorepdarkhelm) [inline app/delivery/send_to_providers.py:29]: PEP-257

File: app/job/rest.py [hunk 6]
```diff
@@ -166,6 +166,10 @@ def get_jobs_by_service(service_id):
 
 @job_blueprint.route("", methods=["POST"])
 def create_job(service_id):
+    """
+    Entry point from UI for one-off messages
+    as well as CSV uploads.
+    """
```
Comment (xlorepdarkhelm) [inline app/job/rest.py:172]: PEP-257

**Review by xlorepdarkhelm — CHANGES_REQUESTED** [review]
Just some PEP-257 requested changes.

File: app/celery/provider_tasks.py (hunk 1 above)
Comment (A-Shumway42) [reply app/celery/provider_tasks.py:96]: @xlorepdarkhelm Oof, it looks like I am breaking some conventions here. Thank you for pointing me to the right place with PEP-257. I'll get these sorted out!

File: app/celery/tasks.py [hunk 7]
```diff
@@ -31,6 +31,7 @@
 
 @notify_celery.task(name="process-job")
 def process_job(job_id, sender_id=None):
+    """ Update job status, get csv data from s3, and begin processing csv rows."""
```
Comment (xlorepdarkhelm) [inline app/celery/tasks.py:34]: space before first word.

**Review by xlorepdarkhelm — APPROVED** [review]
Thanks for getting the PEP-257 updates in., Just one very minor detail (a space before the first word in a comment), otherwise, LGTM. Will approve (please get the little fix, of course).

**Review by ccostino — APPROVED** [review]
Thanks, @A-Shumway42!

**Pull request merged**